llvm/lib/Support/CommandLine.cpp

```diff
@@ -1667,6 +1667,13 @@ bool CommandLineParser::ParseCommandLineOptions(int argc,
       Handler = LookupLongOption(*ChosenSubCommand, ArgName, Value,
                                  LongOptionsUseDoubleDash, HaveDoubleDash);
 
+      // If Handler is not found in a specialized subcommand, look up handler
+      // in the top-level subcommand.
+      // cl::opt without cl::sub belongs to top-level subcommand.
+      if (!Handler && ChosenSubCommand != &SubCommand::getTopLevel())
```

Comment:
Lets split the changes llvm/Support/CommandLine out into a separate patch and add a test to cover this case. We can reference this PR as the motivation.

Reply:
> Lets split the changes llvm/Support/CommandLine out into a separate patch and add a test to cover this case. We can reference this PR as the motivation.

Sent out pr/71776 as a separate patch.